lib/actions/preload.coffee

```diff
@@ -0,0 +1,206 @@
+			select: [ 'id', 'app_name', 'device_type' ]
+			orderby: 'app_name asc'
+	# manual filtering
+	.then (applications) ->
+		applications.forEach (application) ->
+			application.build = application.build.filter (build) ->
+				build.status == 'success'
+		applications
+
+selectApplication = (resin, form, deviceType) ->
+	getApplicationsWithSuccessfulBuilds(resin, deviceType)
+	.then (applications) ->
+		if applications.length == 0
+			expectedError("You have no apps for a '#{deviceType}' device type.")
+		form.ask
+			message: 'Select an application'
+			type: 'list'
+			choices: applications.map (app) ->
+				name: app.app_name
+				value: app
+
+selectApplicationCommit = (resin, form, builds) ->
+	if builds.length == 0
+		expectedError('This application has no successful builds.')
+	DEFAULT_CHOICE = {'name': LATEST, 'value': LATEST}
+	choices = [ DEFAULT_CHOICE ].concat builds.map (build) ->
+		name: "#{build.push_timestamp} - #{build.commit_hash}"
+		value: build.commit_hash
+	return form.ask
+		message: 'Select a build'
+		type: 'list'
+		default: LATEST
+		choices: choices
+
+disableAutomaticUpdates = (form, resin, application, commit) ->
+	if commit == LATEST or not application.should_track_latest_release
+		return Promise.resolve()
+	message = '''
+
+		This application is set to automatically update all devices to the latest
+		available version.
+		You need to disable this if you want to preload a specific build.
+		Do you want to disable automatic updates for this application?
+	'''
+	form.ask
+		message: message,
+		type: 'confirm'
+	.then (update) ->
+		if not update
+			return
+		resin.pine.patch
+			resource: 'application'
+			id: application.id
+			body:
+				should_track_latest_release: false
+
+expectedError = (message) ->
+	console.error(message)
+	process.exit(1)
+
+module.exports =
+	signature: 'preload <image>'
+	description: '(beta) preload an app on a disk image'
+	help: '''
+		Warning: "resin preload" requires Docker to be correctly installed in
+		your shell environment. For more information (including Windows support)
+		please check the README here: https://github.com/resin-io/resin-cli .
+
+		Use this command to preload an application to a local disk image.
+
+		Examples:
+		  $ resin preload resin.img --app 1234 --commit e1f2592fc6ee949e68756d4f4a48e49bff8d72a0 --splash-image some-image.png
+		  $ resin preload resin.img
+	'''
+	permission: 'user'
+	primary: true
+	options: dockerUtils.appendConnectionOptions [
+		{
+			signature: 'app'
+			parameter: 'appId'
+			description: 'id of the application to preload'
+			alias: 'a'
+		}
+		{
+			signature: 'commit'
+			parameter: 'hash'
+			description: 'a specific application commit to preload (ignored if no appId is given)'
+			alias: 'c'
+		}
+		{
+			signature: 'splash-image'
+			parameter: 'splashImage.png'
+			description: 'path to a png image to replace the splash screen'
+			alias: 's'
+		}
+		{
+			signature: 'dont-detect-flasher-type-images'
```

Comment:
This is something that shouldn't be used unless you know what you're doing.
I'd like it to keep a long obscure name.
Discussed here : https://github.com/resin-io/resin-preload/pull/81#discussion_r131387302